lib/internal/modules/esm/default_resolve.js

```diff
@@ -24,6 +25,10 @@ function search(target, base) {
     // We cannot search without a base.
     throw new ERR_MISSING_MODULE(target);
   }
+  if ((target[0] === '.' || target[0] === '/') &&
+  (extname(target) <= 1)) {
```

Comment:
It may be a little clearer to have an explicit `.length` check here instead of relying on JavaScript magic.